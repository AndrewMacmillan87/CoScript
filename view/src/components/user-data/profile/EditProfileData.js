import React from "react";

import PropTypes from "prop-types";

const EditProfileData = ({
    id,
    editName,
    deleteID,
    inputField,
    onChange,
    cancelEdit,
    deleteData,
    onSubmitAbout,
    onSubmitAuthors,
    onSubmitBooks
}) => {
    let editHeading;
    let submitFunc;
    let placeholder;
    switch (id) {
        case "user-about":
            editHeading = "about me";
            submitFunc = onSubmitAbout;
            placeholder = "Write something about yourself...";
            break;
        case "user-authors":
            editHeading = "favourite authors";
            submitFunc = onSubmitAuthors;
            placeholder = "Please separate names with commas...";
            break;
        case "user-books":
            editHeading = "favourite books";
            submitFunc = onSubmitBooks;
            placeholder = "Please separate titles with commas...";
            break;
        default:
            editHeading = "";
    }
    return (
        <div className='profile-edit-container'>
            <form onSubmit={e => submitFunc(e)}>
                <p className='profile-edit-heading'>Edit {editHeading}</p>
                <div className='textarea-container'>
                    <textarea
                        id={id}
                        className='profile-edit-textarea'
                        name={editName}
                        onChange={e => onChange(e)}
                        placeholder={placeholder}
                        ref={inputField}
                    />
                </div>
                <div className='btn-container'>
                    <button className='cancel-btn' onClick={cancelEdit}>
                        Cancel
                    </button>
                    <button
                        id={deleteID}
                        className='delete-btn'
                        onClick={e => deleteData(e)}>
                        Delete
                    </button>
                    <input className='confirm-btn' type='submit' value='Save' />
                </div>
            </form>
        </div>
    );
};

EditProfileData.propTypes = {
    id: PropTypes.string,
    editName: PropTypes.string,
    deleteID: PropTypes.string,
    inputField: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    cancelEdit: PropTypes.func.isRequired,
    deleteData: PropTypes.func.isRequired,
    onSubmitAbout: PropTypes.func.isRequired,
    onSubmitAuthors: PropTypes.func.isRequired,
    onSubmitBooks: PropTypes.func.isRequired
};

export default EditProfileData;
