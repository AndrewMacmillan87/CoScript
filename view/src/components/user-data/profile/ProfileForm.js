import React from "react";
import EditProfileData from "./EditProfileData";

import PropTypes from "prop-types";

const ProfileForm = ({
    user,
    displayEditWindow,
    editField,
    editName,
    deleteID,
    inputField,
    cancelEdit,
    deleteData,
    onChange,
    onClickName,
    onClick,
    onSubmitAbout,
    onSubmitAuthors,
    onSubmitBooks
}) => {
    const editIcon = require("../../../img/edit.svg");
    return (
        <div className='profile-form-container'>
            {displayEditWindow && (
                <EditProfileData
                    id={editField}
                    editName={editName}
                    deleteID={deleteID}
                    inputField={inputField}
                    cancelEdit={cancelEdit}
                    deleteData={deleteData}
                    onChange={onChange}
                    onSubmitAbout={onSubmitAbout}
                    onSubmitAuthors={onSubmitAuthors}
                    onSubmitBooks={onSubmitBooks}
                />
            )}
            <div className='profile-form'>
                <div className='profile-box'>
                    <p>
                        <b>Publicly visible name:</b>
                    </p>
                    <div className='displayed-name'>
                        <input
                            id='username-radio'
                            type='radio'
                            name='name-radio'
                            onClick={e => onClickName(e)}
                        />
                        <label htmlFor='username'>Username</label>
                        <br />
                        <input
                            id='myname-radio'
                            type='radio'
                            name='name-radio'
                            onClick={e => onClickName(e)}
                        />
                        <label htmlFor='myname'>My name</label>
                    </div>
                </div>
                <div className='profile-box'>
                    <label className='profile-label' htmlFor='user-about'>
                        About Me:
                    </label>
                    <div className='profile-data-container'>
                        <div className='profile-data'>
                            <ul className='data-display'>
                                {user && user.profile.about}
                            </ul>
                        </div>
                        <button className='edit-icon-btn' type='button'>
                            <img
                                className='edit-input-icon'
                                name='about'
                                onClick={e => onClick(e)}
                                src={editIcon}
                                alt='Edit input icon'
                            />
                        </button>
                    </div>
                </div>
                <div className='profile-box'>
                    <label className='profile-label' htmlFor='user-authors'>
                        Favourite Authors:
                    </label>
                    <div className='profile-data-container'>
                        <div className='profile-data'>
                            <ul className='data-display'>
                                {user &&
                                    user.profile.authors &&
                                    user.profile.authors.map(item => (
                                        <li
                                            className='profile-list-item'
                                            key={item}>
                                            {item}
                                        </li>
                                    ))}
                            </ul>
                        </div>
                        <button className='edit-icon-btn' type='button'>
                            <img
                                className='edit-input-icon'
                                name='authors'
                                onClick={e => onClick(e)}
                                src={editIcon}
                                alt='Edit input icon'
                            />
                        </button>
                    </div>
                </div>
                <div className='profile-box'>
                    <label className='profile-label' htmlFor='user-books'>
                        Favourite Books:
                    </label>
                    <div className='profile-data-container'>
                        <div className='profile-data'>
                            <ul className='data-display'>
                                {user &&
                                    user.profile.books &&
                                    user.profile.books.map(item => (
                                        <li
                                            className='profile-list-item'
                                            key={item}>
                                            {item}
                                        </li>
                                    ))}
                            </ul>
                        </div>
                        <button className='edit-icon-btn' type='button'>
                            <img
                                className='edit-input-icon'
                                name='books'
                                onClick={e => onClick(e)}
                                src={editIcon}
                                alt='Edit input icon'
                            />
                        </button>
                    </div>
                </div>
            </div>
        </div>
    );
};

ProfileForm.propTypes = {
    user: PropTypes.object,
    displayEditWindow: PropTypes.bool.isRequired,
    editField: PropTypes.string.isRequired,
    editName: PropTypes.string.isRequired,
    deleteID: PropTypes.string.isRequired,
    inputField: PropTypes.object.isRequired,
    cancelEdit: PropTypes.func.isRequired,
    deleteData: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    onClickName: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    onSubmitAbout: PropTypes.func.isRequired,
    onSubmitAuthors: PropTypes.func.isRequired,
    onSubmitBooks: PropTypes.func.isRequired
};

export default ProfileForm;
