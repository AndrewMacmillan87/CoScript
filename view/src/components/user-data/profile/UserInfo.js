import React, { Fragment } from "react";

import Alert from "../../alert/Alert";
import ProfileForm from "./ProfileForm";
import MyProjects from "./MyProjects";
import Collaborations from "./Collaborations";

import PropTypes from "prop-types";

const UserInfo = ({
    user,
    displayEditWindow,
    editField,
    editName,
    deleteID,
    projects,
    userCollaborations,
    history,
    projectTitle,
    projectGenres,
    projectDescription,
    displayProjectEditWindow,
    displayDeleteProjectWindow,
    openDeleteProjectWindow,
    removeDeleteProjectWindow,
    displayCollaborateDeleteWindow,
    openCollaborateDeleteWindow,
    removeCollaborateDeleteWindow,
    onDeleteProject,
    onDeleteCollaborate,
    onClickProject,
    onSubmitProject,
    saving,
    savingCollaborate,
    loading,
    inputField,
    onChange,
    onClick,
    cancelEdit,
    deleteData,
    onClickName,
    onSubmitAbout,
    onSubmitAuthors,
    onSubmitBooks,
}) => {
    const userImage = require("../../../img/user.svg");
    return (
        <Fragment>
            <Alert />
            <div className='user-info-container'>
                <h1 className='profile-h1'>My Profile</h1>
                <section className='profile'>
                    <h2 className='profile-heading' id='profile-h2'>
                        Profile
                    </h2>
                    <ProfileForm
                        user={user}
                        displayEditWindow={displayEditWindow}
                        editField={editField}
                        editName={editName}
                        deleteID={deleteID}
                        inputField={inputField}
                        onChange={onChange}
                        onClick={onClick}
                        cancelEdit={cancelEdit}
                        deleteData={deleteData}
                        onClickName={onClickName}
                        onSubmitAbout={onSubmitAbout}
                        onSubmitAuthors={onSubmitAuthors}
                        onSubmitBooks={onSubmitBooks}
                    />
                </section>
                <section className='photo'>
                    <h2 className='profile-heading' id='photo-h2'>
                        Photo
                    </h2>
                    <div className='change-avatar-container'>
                        <img
                            className='avatar'
                            src={user ? user.avatar : userImage}
                            alt='Users profile avatar'
                        />
                        <div className='profile-image-text-container'>
                            <p>Want a different image?</p>
                            <label
                                className='profile-image-text'
                                htmlFor='profile-image-upload'>
                                Click here
                                <input
                                    className='choose-file'
                                    id='profile-image-upload'
                                    type='file'
                                />
                            </label>
                            <p> to upload a new one</p>
                        </div>
                        <br />
                    </div>
                </section>
                <section className='my-projects'>
                    <h2 className='profile-heading' id='projects-h2'>
                        My Projects
                    </h2>
                    <MyProjects
                        user={user}
                        projects={projects}
                        projectTitle={projectTitle}
                        projectGenres={projectGenres}
                        projectDescription={projectDescription}
                        displayProjectEditWindow={displayProjectEditWindow}
                        displayDeleteProjectWindow={displayDeleteProjectWindow}
                        openDeleteProjectWindow={openDeleteProjectWindow}
                        removeDeleteProjectWindow={removeDeleteProjectWindow}
                        onChange={onChange}
                        cancelEdit={cancelEdit}
                        onDeleteProject={onDeleteProject}
                        onClickProject={onClickProject}
                        onSubmitProject={onSubmitProject}
                        saving={saving}
                        loading={loading}
                    />
                </section>
                <section className='starred-projects'>
                    <h2 className='profile-heading' id='starred-projects-h2'>
                        Collaborations
                    </h2>
                    <Collaborations
                        user={user}
                        userCollaborations={userCollaborations}
                        displayCollaborateDeleteWindow={
                            displayCollaborateDeleteWindow
                        }
                        openCollaborateDeleteWindow={
                            openCollaborateDeleteWindow
                        }
                        removeCollaborateDeleteWindow={
                            removeCollaborateDeleteWindow
                        }
                        onDeleteCollaborate={onDeleteCollaborate}
                        savingCollaborate={savingCollaborate}
                        onChange={onChange}
                        history={history}
                        loading={loading}
                    />
                </section>
            </div>
        </Fragment>
    );
};

UserInfo.propTypes = {
    user: PropTypes.object,
    displayEditWindow: PropTypes.bool.isRequired,
    editField: PropTypes.string.isRequired,
    editName: PropTypes.string.isRequired,
    deleteID: PropTypes.string.isRequired,
    projects: PropTypes.array,
    userCollaborations: PropTypes.array,
    history: PropTypes.object,
    projectTitle: PropTypes.string,
    projectGenres: PropTypes.string,
    projectDescription: PropTypes.string,
    displayProjectEditWindow: PropTypes.bool.isRequired,
    displayDeleteProjectWindow: PropTypes.bool.isRequired,
    openDeleteProjectWindow: PropTypes.func.isRequired,
    removeDeleteProjectWindow: PropTypes.func.isRequired,
    onDeleteProject: PropTypes.func.isRequired,
    displayCollaborateDeleteWindow: PropTypes.bool.isRequired,
    openCollaborateDeleteWindow: PropTypes.func.isRequired,
    removeCollaborateDeleteWindow: PropTypes.func.isRequired,
    onDeleteCollaborate: PropTypes.func.isRequired,
    onClickProject: PropTypes.func.isRequired,
    onSubmitProject: PropTypes.func.isRequired,
    saving: PropTypes.bool,
    loading: PropTypes.bool,
    savingCollaborate: PropTypes.bool,
    inputField: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    cancelEdit: PropTypes.func.isRequired,
    deleteData: PropTypes.func.isRequired,
    onClickName: PropTypes.func.isRequired,
    onSubmitAbout: PropTypes.func.isRequired,
    onSubmitAuthors: PropTypes.func.isRequired,
    onSubmitBooks: PropTypes.func.isRequired,
};

export default UserInfo;
