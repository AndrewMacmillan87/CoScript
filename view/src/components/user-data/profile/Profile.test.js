import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../../redux/store/reduxStore";

import { Profile } from "./Profile";
import MobileNav from "../MobileNav";
import SidePanel from "../SidePanel";
import UserInfo from "./UserInfo";
import ProfileForm from "./ProfileForm";
import MyProjects from "./MyProjects";
import Collaborations from "./Collaborations";
import EditProfileData from "./EditProfileData";

const userData = require("../../../redux/tests/actions/data/auth/user.json");
const projectArray = require("../../../redux/tests/actions/data/documents/projectArray.json");

describe("Profile component", () => {
    let props,
        mockUpdateUserProfile,
        mockLoadAllProjects,
        mockUpdateDetails,
        mockGetCollaborations,
        mockClearFileData,
        mockDeleteProject,
        mockDeleteCollaborator;
    beforeEach(() => {
        global.HTMLMediaElement.prototype.pause = jest.fn();
        global.scrollTo = jest.fn();
        mockUpdateUserProfile = jest.fn();
        mockLoadAllProjects = jest.fn();
        mockUpdateDetails = jest.fn();
        mockGetCollaborations = jest.fn();
        mockClearFileData = jest.fn();
        mockDeleteProject = jest.fn();
        mockDeleteCollaborator = jest.fn();
        props = {
            user: userData,
            projectArray: [],
            userCollaborationsArray: [],
            updateUserProfile: mockUpdateUserProfile,
            loadAllProjects: mockLoadAllProjects,
            updateProjectDetails: mockUpdateDetails,
            getUserCollaborations: mockGetCollaborations,
            clearFileData: mockClearFileData,
            deleteProject: mockDeleteProject,
            deleteCollaborator: mockDeleteCollaborator,
        };
    });

    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Profile {...props} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a main element with a className of 'profile-container'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
        expect(wrapper.find(".profile-container").length).toEqual(1);
    });

    it("should render the MobileNave component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(MobileNav).length).toEqual(1);
    });

    it("should render the SidePanel component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(SidePanel).length).toEqual(1);
    });

    it("should render the UserInfo component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(UserInfo).length).toEqual(1);
    });

    it("should call `clearFileData()`, `loadAllProjects()` and `getUserCollaborations()` on mount", async () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        expect(mockClearFileData).toBeCalled();
        await expect(mockLoadAllProjects).toBeCalled();
        expect(mockGetCollaborations).toBeCalled();
    });

    it("should call `updateUserProfile()` when the user updates their displayed name", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        wrapper.find("#username-radio").simulate("click");
        expect(mockUpdateUserProfile).toBeCalled();
    });

    it("should call `loadAllProjects()` and `updateProjectDetails()` when user edits a project's details", async () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayProjectEditWindow: true });
        wrapper.find(".edit-project-form").simulate("submit", { preventDefault: jest.fn() });
        await expect(mockLoadAllProjects).toBeCalled();
        expect(mockUpdateDetails).toBeCalled();
    });

    it("should call `deleteCollaborator()`", async () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayCollaborateDeleteWindow: true });
        wrapper.find(".edit-project-form").at(0).simulate("submit", { preventDefault: jest.fn() });
        await expect(mockDeleteCollaborator).toBeCalled();
    });

    it("should call `deleteProject()`", async () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...props} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayDeleteProjectWindow: true });
        wrapper.find(".edit-project-form").at(0).simulate("submit", { preventDefault: jest.fn() });
        await expect(mockDeleteProject).toBeCalled();
    });
});

describe("UserInfo component", () => {
    let props;
    beforeEach(() => {
        props = {
            user: userData,
            displayEditWindow: false,
            editField: "",
            editName: "",
            deleteID: "",
            projects: [],
            userCollaborations: [],
            history: {},
            projectTitle: "",
            projectGenres: "",
            projectDescription: "",
            displayProjectEditWindow: false,
            displayDeleteProjectWindow: false,
            openDeleteProjectWindow: jest.fn(),
            removeDeleteProjectWindow: jest.fn(),
            onDeleteProject: jest.fn(),
            displayCollaborateDeleteWindow: false,
            openCollaborateDeleteWindow: jest.fn(),
            removeCollaborateDeleteWindow: jest.fn(),
            onDeleteCollaborate: jest.fn(),
            onClickProject: jest.fn(),
            onSubmitProject: jest.fn(),
            saving: false,
            loading: false,
            savingCollaborate: false,
            inputField: {},
            onChange: jest.fn(),
            onClick: jest.fn(),
            cancelEdit: jest.fn(),
            deleteData: jest.fn(),
            onClickName: jest.fn(),
            onSubmitAbout: jest.fn(),
            onSubmitAuthors: jest.fn(),
            onSubmitBooks: jest.fn(),
        };
    });

    afterAll(() => {
        props.projects = [];
    });

    it("should render the ProfileForm component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(ProfileForm).length).toEqual(1);
    });

    it("should render the MyProjects component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(MyProjects).length).toEqual(1);
    });

    it("should render the Collaborations component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Collaborations).length).toEqual(1);
    });

    it("should render the 4 section elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("section").length).toEqual(4);
    });

    it("should render a h1 element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("h1").length).toEqual(1);
    });

    it("should render 4 h2 elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <UserInfo {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("h2").length).toEqual(4);
    });
});

describe("ProfileForm component", () => {
    let profileProps, props;
    beforeEach(() => {
        profileProps = {
            user: userData,
            projectArray: [],
            userCollaborationsArray: [],
            updateUserProfile: jest.fn(),
            loadAllProjects: jest.fn(),
            updateProjectDetails: jest.fn(),
            getUserCollaborations: jest.fn(),
            clearFileData: jest.fn(),
            deleteProject: jest.fn(),
            deleteCollaborator: jest.fn(),
        };
        props = {
            user: userData,
            displayEditWindow: false,
            editField: "",
            editName: "",
            deleteID: "",
            inputField: {},
            cancelEdit: jest.fn(),
            deleteData: jest.fn(),
            onChange: jest.fn(),
            onClickName: jest.fn(),
            onClick: jest.fn(),
            onSubmitAbout: jest.fn(),
            onSubmitAuthors: jest.fn(),
            onSubmitBooks: jest.fn(),
        };
    });

    it("should render the EditProfileData component if 'displayEditWindow' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayEditWindow: true });
        expect(wrapper.find(EditProfileData).length).toEqual(1);
    });

    it("should render a div with a className of 'profile-form-container'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(".profile-form-container").length).toEqual(1);
    });

    it("should render 17 div elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("div").length).toEqual(17);
    });

    it("should render 6 label elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );

        expect(wrapper.find("label").length).toEqual(6);
    });

    it("should render 3 input elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("input").length).toEqual(3);
    });

    it("should render 3 ul elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("ul").length).toEqual(3);
    });

    it("should render 4 button elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );

        expect(wrapper.find("button").length).toEqual(4);
    });

    it("should render 7 img elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );

        expect(wrapper.find("img").length).toEqual(7);
    });
});

describe("MyProjects component", () => {
    let profileProps, props;
    beforeEach(() => {
        profileProps = {
            user: userData,
            projectArray: [],
            userCollaborationsArray: [],
            updateUserProfile: jest.fn(),
            loadAllProjects: jest.fn(),
            updateProjectDetails: jest.fn(),
            getUserCollaborations: jest.fn(),
            clearFileData: jest.fn(),
            deleteProject: jest.fn(),
            deleteCollaborator: jest.fn(),
        };
        props = {
            user: userData,
            projects: [],
            projectTitle: "",
            projectGenres: "",
            projectDescription: "",
            displayProjectEditWindow: false,
            displayDeleteProjectWindow: false,
            openDeleteProjectWindow: jest.fn(),
            removeDeleteProjectWindow: jest.fn(),
            onChange: jest.fn(),
            cancelEdit: jest.fn(),
            onDeleteProject: jest.fn(),
            onClickProject: jest.fn(),
            onSubmitProject: jest.fn(),
            saveDate: jest.fn(),
            saving: false,
            loading: false,
            clearFileData: jest.fn(),
            clearProjectData: jest.fn(),
        };
    });

    it("should render a div element with a className of 'delete-project-window' if 'displayEditWindow' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayDeleteProjectWindow: true });
        expect(wrapper.find(".delete-project-window").length).toEqual(1);
    });

    it("should render a div element with a className of 'edit-project-details-window' if 'displayProjectEditWindow' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayProjectEditWindow: true });
        expect(wrapper.find(".edit-project-details-window").length).toEqual(1);
    });

    it("should render 2 img elements with a className of 'profile-project-loading-spinner' if 'loading' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ loading: true });
        expect(wrapper.find(".profile-project-loading-spinner").length).toEqual(2);
    });

    it("should render a div with a className of 'projects-warning' if the user has no projects", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <MyProjects {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(".projects-warning").length).toEqual(1);
    });
});

describe("Collaborations component", () => {
    let profileProps, props;
    beforeEach(() => {
        profileProps = {
            user: userData,
            projectArray: [],
            userCollaborationsArray: [],
            updateUserProfile: jest.fn(),
            loadAllProjects: jest.fn(),
            updateProjectDetails: jest.fn(),
            getUserCollaborations: jest.fn(),
            clearFileData: jest.fn(),
            deleteProject: jest.fn(),
            deleteCollaborator: jest.fn(),
        };
        props = {
            user: userData,
            userCollaborations: [],
            openCollaborateDeleteWindow: jest.fn(),
            removeCollaborateDeleteWindow: jest.fn(),
            onDeleteCollaborate: jest.fn(),
            savingCollaborate: false,
            loading: false,
            onChange: jest.fn(),
            loadProject: jest.fn(),
        };
    });

    it("should render a div element with a className of 'delete-project-window' if 'displayCollaborateDeleteWindow' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ displayCollaborateDeleteWindow: true });
        expect(wrapper.find(".delete-project-window").length).toEqual(1);
    });

    it("should render 2 img elements with a className of 'profile-project-loading-spinner' if 'loading' is true", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Profile {...profileProps} />
                </Router>
            </Provider>
        );
        wrapper.find(Profile).setState({ loading: true });
        expect(wrapper.find(".profile-project-loading-spinner").length).toEqual(2);
    });
});
