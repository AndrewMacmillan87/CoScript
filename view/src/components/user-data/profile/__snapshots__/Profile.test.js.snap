// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
<main
  className="profile-container"
>
  <nav
    className="profile-mobile-nav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="home-icon fas fa-home fa-3x"
      />
    </a>
    <button
      className="profile-dropdown-icon-link"
      onClick={[Function]}
    >
      <img
        alt="Burger menu icon"
        className="profile-dropdown-icon"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "menu.svg",
          }
        }
      />
    </button>
    <div
      className="dropdown-menu dropdown-content"
      id="dropdown-nav"
    >
      <a
        href="/profile"
        onClick={[Function]}
      >
        <p
          className="dropdown-link"
        >
          My Profile
        </p>
      </a>
      <a
        href="/messages"
        onClick={[Function]}
      >
        <p
          className="dropdown-link"
        >
          Messages
        </p>
      </a>
      <a
        href="/projects"
        onClick={[Function]}
      >
        <p
          className="dropdown-link"
        >
          Active Projects
        </p>
      </a>
      <a
        href="/publications"
        onClick={[Function]}
      >
        <p
          className="dropdown-link"
        >
          Publications
        </p>
      </a>
      <a
        href=" "
        onClick={[Function]}
      >
        <p
          className="dropdown-link"
        >
          Logout
        </p>
      </a>
    </div>
  </nav>
  <div
    className="user-info-container"
  >
    <h1
      className="profile-h1"
    >
      My Profile
    </h1>
    <section
      className="profile"
    >
      <h2
        className="profile-heading"
        id="profile-h2"
      >
        Profile
      </h2>
      <div
        className="profile-form-container"
      >
        <div
          className="profile-form"
        >
          <div
            className="profile-box"
          >
            <p>
              <b>
                Publicly visible name:
              </b>
            </p>
            <div
              className="displayed-name"
            >
              <input
                id="username-radio"
                name="name-radio"
                onClick={[Function]}
                type="radio"
              />
              <label
                htmlFor="username"
              >
                Username
              </label>
              <br />
              <input
                id="myname-radio"
                name="name-radio"
                onClick={[Function]}
                type="radio"
              />
              <label
                htmlFor="myname"
              >
                My name
              </label>
            </div>
          </div>
          <div
            className="profile-box"
          >
            <label
              className="profile-label"
              htmlFor="user-about"
            >
              About Me:
            </label>
            <div
              className="profile-data-container"
            >
              <div
                className="profile-data"
              >
                <ul
                  className="data-display"
                />
              </div>
              <button
                className="edit-icon-btn"
                type="button"
              >
                <img
                  alt="Edit input icon"
                  className="edit-input-icon"
                  name="about"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "edit.svg",
                    }
                  }
                />
              </button>
            </div>
          </div>
          <div
            className="profile-box"
          >
            <label
              className="profile-label"
              htmlFor="user-authors"
            >
              Favourite Authors:
            </label>
            <div
              className="profile-data-container"
            >
              <div
                className="profile-data"
              >
                <ul
                  className="data-display"
                />
              </div>
              <button
                className="edit-icon-btn"
                type="button"
              >
                <img
                  alt="Edit input icon"
                  className="edit-input-icon"
                  name="authors"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "edit.svg",
                    }
                  }
                />
              </button>
            </div>
          </div>
          <div
            className="profile-box"
          >
            <label
              className="profile-label"
              htmlFor="user-books"
            >
              Favourite Books:
            </label>
            <div
              className="profile-data-container"
            >
              <div
                className="profile-data"
              >
                <ul
                  className="data-display"
                />
              </div>
              <button
                className="edit-icon-btn"
                type="button"
              >
                <img
                  alt="Edit input icon"
                  className="edit-input-icon"
                  name="books"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "edit.svg",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="photo"
    >
      <h2
        className="profile-heading"
        id="photo-h2"
      >
        Photo
      </h2>
      <div
        className="change-avatar-container"
      >
        <img
          alt="Users profile avatar"
          className="avatar"
          src="www.gravatar.com/avatar/123456789"
        />
        <div
          className="profile-image-text-container"
        >
          <p>
            Want a different image?
          </p>
          <label
            className="profile-image-text"
            htmlFor="profile-image-upload"
          >
            Click here
            <input
              className="choose-file"
              id="profile-image-upload"
              type="file"
            />
          </label>
          <p>
             to upload a new one
          </p>
        </div>
        <br />
      </div>
    </section>
    <section
      className="my-projects"
    >
      <h2
        className="profile-heading"
        id="projects-h2"
      >
        My Projects
      </h2>
      <img
        alt="loading"
        className="profile-project-loading-spinner"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "loading-black.svg",
          }
        }
      />
    </section>
    <section
      className="starred-projects"
    >
      <h2
        className="profile-heading"
        id="starred-projects-h2"
      >
        Collaborations
      </h2>
      <img
        alt="loading"
        className="profile-project-loading-spinner"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "loading-black.svg",
          }
        }
      />
    </section>
  </div>
</main>
`;
